[builtins] adds ToggleRegionVisualization builtin

mkortstiege opened the pull request:
Adds new `ToggleDirtyRegions` builtin. This enables on-the-fly dirty-region visualization toggling without the need to restart Kodi.

Requested by @HitcherUK.

HitcherUK commented:
Thanks.

mkortstiege commented:
@Montellese objections?

mkortstiege commented:
Will shove this in later today if there are no objections.

Montellese commented:
Could you rename the builtin to make it clear that it toggles the visualization? `ToggleDirtyRegions` sounds to me like it switches between two different modes of dirty regions (even though there are at least three).

mkortstiege commented:
Sure. What about 'ToggleDirtyRegionVisualization` or just `ToggleDRVisualization`? A bit long maybe ..

mkortstiege committed:
[builtins] adds ToggleDirtyRegionVisualization builtin

mkortstiege commented:
Made it `ToggleRegionVisualization` now.

mkortstiege committed:
[keymap] let ctrl+shift+r toggle dirty region visualization

Montellese commented:
I'd go with `ToggleDirtyRegionVisualization` as it explains best what it does. `ToggleRegionVisualization` is not that obvious by itself without any documentation.

topfs2 commented:
Im guessing that the original advanced settings is because not all visualisations understand dirty regions? Kind of outside this PR but wouldn't it be better if they simply flagged their support in the add-on.XML, removing the need for this?

mkortstiege commented:
@Montellese. Will do.
@topfs2 it's not about visualizations ;) This is is to toggle dirty region visualization on/off without the need to restart Kodi.

topfs2 commented:
Oh wait. Is this the debug were it displays them on screen?

topfs2 commented:
Do we name debug stuff like this in any consistent matter. Would be nice if you instantly knew it is debug related. Perhaps something like DebugShowDirtyRegions

mkortstiege commented:
Updated. Builtin now uses `ToggleDirtyRegionVisualization`. I've also added a keyboard keymap commit that enables this via `ctrl+shift+r` (analogue to ToggleDebug).

@topfs2, I'd like to keep the term Toggle in there without making it longer.

Montellese commented:
jenkins build and merge

topfs2 commented:
@mkortstiege yeah probably good with toggle naming tbh
